### PR TITLE
Add new `--sourceMaps` option in CLI

### PR DESCRIPTION
**Pull request by matuzalemsteles**
hey guys, I've been working on [generating source maps for the soy files](https://github.com/matuzalemsteles/generate-soy-source-maps) in the last few days. It seems to be very promising to improve the developer's experience in debugging in development environment and testing coverage.

We still have some work to do in `soyparser` to have a better experience and to be able to map in the best way.

### Currently supported types

* Call
* DelTemplate
* LetStatement
* Param
* ParamDeclaration
* Template

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
cc @jbalsas

File: bin/metalsoy.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ const yargs = require('yargs')
 	})
 	.option('externalMsgFormat', {
 		default: defaultOptions.externalMsgFormat,
-		describe: 'Pattern that should be used to format the extracted external messages from compiled files',
+		describe:
+			'Pattern that should be used to format the extracted external messages from compiled files',
```
Comment (jbalsas) [inline bin/metalsoy.js:18]: Is this automatically formatted? 🤔

File: lib/pipelines/__tests__/compileSoy.js [hunk 2]
```diff
@@ -258,6 +259,47 @@ describe('Compile Soy Pipeline', function() {
 		});
 	});
 
+	it('should not add the source map path to the end of the generated file when skipMetalGeneration is true', function(
```
Comment (jbalsas) [inline lib/pipelines/__tests__/compileSoy.js]: Why is this the case? Wouldn't we still want to be able to debug the client-side template even if we're not generating a metal component for it? Or is it that it's not really something we're using?

**Review by jbalsas — CHANGES_REQUESTED** [review]
Hey @matuzalemsteles, this looks great! I've just added a comment regarding the `skipMetalGeneration` and why we don't want to generate sourcemaps for it.

Other than that, we should be looking into transferring the sourcemaps generator into the metal org as well... maybe even inside metal-tools-soy so we don't even need to publish it independently... or turning this into a lerna repo to hold different packages... what do you think?

File: bin/metalsoy.js (hunk 1 above)
Comment (matuzalemsteles) [reply bin/metalsoy.js:18]: hey @jbalsas, when I did the commit, it ended up updating. 😁

File: lib/pipelines/__tests__/compileSoy.js (hunk 2 above)
Comment (matuzalemsteles) [reply lib/pipelines/__tests__/compileSoy.js]: hey @jbalsas, I have not had the opportunity to test this without skipMetalGeneration ... But currently I do not see using that somewhere. (I can be mistaken 😅)

I'm going to remove this and go with the possibility of generating the independent mapping of `skipMetalGeneration`.

**Comment by matuzalemsteles** [comment]
hey @jbalsas, Great idea. We can make **metal-tools-soy** a repo lerna. Would it be interesting to move the `soyparser` and `metal-soy-critic` to here as well?

**Comment by jbalsas** [comment]
Hey @matuzalemsteles, I've merged this to `develop`. Do you want us to make a release with this, or should we wait until the `soyparser` is updated and maybe until we move here all the soy-related packages?

**Comment by matuzalemsteles** [comment]
hey @jbalsas thanks. I think we can do a launch with the implementation, so people can start providing feedback... This does not yet depend on the changes of `soyparser`... We can then continue with these changes to be a repo lerna.

I think we can follow a timeline after that, something like:

* Release metal-tools-soy
* Merge PR https://github.com/metal/soyparser/pull/4
* Release **soyparser**
* Aggregate Soy Tooling in this repo #35
* Improves **soyparser** API with Soy grammar

What do you think?

**Comment by jbalsas** [comment]
Hey @matuzalemsteles, just double-checking... now that metal/soyparser#4 has been merged... what do you think of waiting to get this all out the door together?

**Comment by matuzalemsteles** [comment]
hey @jbalsas, Sure, no problem. As soon as we have a release of `soyparser` I will add `delcall` type support in **generate-soy-source-maps**, to see in action.

**Comment by jbalsas** [comment]
Awesome! Excited about this! :)

**Pull request merged**